gem/tests/test_forms.py

```diff
@@ -0,0 +1,55 @@
+from django.test import TestCase
+from django.contrib.auth.models import User
+
+from gem.forms import RegistrationForm
+
+
+class RegisterTestCase(TestCase):
+
+    def test_register_username_correct(self):
+        form_data = {
+            'username': 'Jeyabal@-1',
+            'password': '1234',
+            'gender': 'f',
+        }
+        form = RegistrationForm(data=form_data)
+        self.assertEqual(form.is_valid(), True)
+
+    def test_register_username_incorrect(self):
+        form_data = {
+            'username': 'Jeyabal#',
+            'password': '1234',
+            'gender': 'f',
+        }
+        form = RegistrationForm(data=form_data)
+        self.assertEqual(form.is_valid(), False)
+
+    def test_register_password_incorrect(self):
+        form_data = {
+            'username': 'Jeyabal#',
```

Comment:
By using a bad username here, the form will be incorrect, even if the password is correct.